src/System.Diagnostics.Process/tests/System.Diagnostics.Process.Tests/ProcessStreamReadTests.cs

```diff
@@ -130,19 +135,44 @@ public void TestAsyncHalfCharacterAtATime()
             p.StartInfo.StandardOutputEncoding = Encoding.Unicode;
             p.OutputDataReceived += (s, e) =>
             {
-                Assert.Equal(e.Data, "a");
+                if (!receivedOutput)
+                {
+                    Assert.Equal(e.Data, "a");
+                }
                 receivedOutput = true;
             };
             p.Start();
             p.BeginOutputReadLine();
 
-            if (p.WaitForExit(WaitInMS))
-                p.WaitForExit(); // This ensures async event handlers are finished processing.
+            Assert.True(p.WaitForExit(WaitInMS));
+            p.WaitForExit(); // This ensures async event handlers are finished processing.
 
             Assert.True(receivedOutput);
         }
 
         [Fact]
+        public void TestManyOutputLines()
+        {
+            int linesReceived = 0;
+            Process p = CreateProcessManyOutputLines();
+            p.StartInfo.RedirectStandardOutput = true;
+            p.OutputDataReceived += (s, e) =>
+            {
+                if (!string.IsNullOrEmpty(e.Data))
+                {
+                    linesReceived++;
+                }
+            };
+            p.Start();
+            p.BeginOutputReadLine();
+
+            Assert.True(p.WaitForExit(WaitInMS));
+            p.WaitForExit(); // This ensures async event handlers are finished processing.
+
+            Assert.Equal(144, linesReceived);
```

Comment:
Could you also add total callback invocation counter and assert for him
`Assert.Equal(145, callbackInvocationCounter);`
for ensuring that we receive trailing null.